scripts/main.js

```diff
@@ -0,0 +1,190 @@
+  const reader = new FileReader();
+
+  reader.addEventListener(
+    "load",
+    () => {
+      imagePreview.src = reader.result;
+    },
+    false
+  );
+
+  reader.readAsDataURL(selectedImageFile);
+}
+
+// ---
+
+function getCaptionElement() {
+  if (state.caption) {
+    const captionElement = document.createElement("p");
+    captionElement.innerText = `"${state.caption}"`;
+    return captionElement;
+  }
+}
+
+// ---
+
+function getCopyButton() {
+  const button = document.createElement("button");
+  button.setAttribute("type", "button");
+  button.innerText = "Copy caption text";
+
+  button.addEventListener("click", onCopyButtonClicked);
+
+  return button;
+}
+
+// ---
+
+function getImgCodeSnippet() {
+  const imagePreviewElement = document.querySelector("#image-preview");
+  const imageSource = imagePreviewElement.getAttribute("src");
+  const codeSnippet = `<img alt="${state.caption}" src="${imageSource}">`;
+  return codeSnippet;
+}
+
+// ---
+
+function getCopyImgCodeSnippetButton() {
+  const button = document.createElement("button");
+  button.setAttribute("type", "button");
+  button.innerText = "Copy code snippet";
+
+  button.addEventListener("click", onCopyImgCodeSnippetButtonClicked);
+
+  return button;
+}
+
+// ---* for event listeners *---
+function onCopyImgCodeSnippetButtonClicked() {
+  navigator.clipboard.writeText(state.codeSnippet);
+}
+
+function onCopyButtonClicked() {
+  navigator.clipboard.writeText(state.caption);
+}
+
+function onImageInputChanged() {
+  const selectedImageFile = imageInputElement.files[0];
+  state.image = selectedImageFile.name;
+
+  renderImagePreview(selectedImageFile);
+  renderGetCaptionButton();
+  const getCaptionButton = document.querySelector("#get-caption");
+  getCaptionButton.addEventListener("click", onGetCaptionButtonClicked);
+}
+
+function onGetCaptionButtonClicked() {
+  requestForAPI();
+}
+
+//---
+
+// fetch request
+async function requestForAPI() {
+  const selectedImageFile = imageInputElement.files[0];
+  const data = new FormData();
+  data.append("imageFile", selectedImageFile, "file");
+
+  const fetchHeader = new Headers();
+  fetchHeader.append("Apikey", "eea0fd3c-6342-4543-a92c-9fc111175b97");
+
+  const options = {
+    method: "POST",
+    body: data,
+    credentials: "omit",
+    headers: fetchHeader,
+  };
+
+  await fetch("https://api.cloudmersive.com/image/recognize/describe", options)
+    .then(function (response) {
+      // The response is a Response instance.
+      // You parse the data into a useable format using `.json()`
+      return response.json();
+    })
+    .then(function (data) {
+      // `data` is the parsed version of the JSON returned from the above endpoint.
+      state.fetchedJson = data;
+      state.caption = state.fetchedJson.BestOutcome.Description;
+      state.codeSnippet = getImgCodeSnippet();
+    })
+    .then(renderCaptionElement)
+    .then(renderCopyButton)
+    .then(renderCopyImgCodeSnippetButton);
```

Comment:
Because of await, you don't need to then chain like this anymore
```suggestion
const response = await fetch("https://api.cloudmersive.com/image/recognize/describe", options)

// The response is a Response instance.
// You parse the data into a useable format using `.json()`
state.fetchedJson = await response.json();
state.caption = state.fetchedJson.BestOutcome.Description;
state.codeSnippet = getImgCodeSnippet();
renderCaptionElement();
renderCopyButton();
renderCopyImgCodeSnippetButton();
```

Reply:
I forgot you could use multiple await keywords this way in an async function! Awesome and makes sense. Thank you!